test/steps/configuration_steps.go

```diff
@@ -35,27 +35,24 @@ func ConfigurationSteps(suite *godog.Suite, fs *FeatureState) {
 		return nil
 	})
 
-	suite.Step(`^the global new-branch-push-flag configuration is false$`, func() error {
-		_ = fs.activeScenarioState.gitEnvironment.DeveloperRepo.Configuration(false).SetNewBranchPush(false, true)
-		return nil
-	})
-
-	suite.Step(`^the new-branch-push-flag configuration is now false$`, func() error {
-		if fs.activeScenarioState.gitEnvironment.DeveloperRepo.Configuration(true).ShouldNewBranchPush() {
-			return fmt.Errorf("expected global new-branch-push-flag to be false, but was true")
+	suite.Step(`^the new-branch-push-flag configuration is now (true|false)$`, func(text string) error {
+		want, err := strconv.ParseBool(text)
+		if err != nil {
+			return fmt.Errorf("cannot parse %q into bool: %w", text, err)
 		}
-		return nil
-	})
-
-	suite.Step(`^the new-branch-push-flag configuration is now true$`, func() error {
-		if !fs.activeScenarioState.gitEnvironment.DeveloperRepo.Configuration(true).ShouldNewBranchPush() {
-			return fmt.Errorf("expected global new-branch-push-flag to be true, but was false")
+		have := fs.activeScenarioState.gitEnvironment.DeveloperRepo.Configuration(true).ShouldNewBranchPush()
+		if have != want {
+			return fmt.Errorf("expected global new-branch-push-flag to be %t, but was %t", want, have)
 		}
 		return nil
 	})
 
-	suite.Step(`^the global new-branch-push-flag configuration is true$`, func() error {
-		_ = fs.activeScenarioState.gitEnvironment.DeveloperRepo.Configuration(false).SetNewBranchPush(true, true)
+	suite.Step(`^the global new-branch-push-flag configuration is (true|false)$`, func(text string) error {
+		b, err := strconv.ParseBool(text)
+		if err != nil {
+			return fmt.Errorf("cannot parse %q into bool: %w", text, err)
```

Comment:
We can just return `err` and ParseBool returns a good error message that includes the text passed in

Reply:
Good catch!